src/main/scala/exercises/06_Collections.scala

```diff
@@ -130,32 +115,36 @@ object Collections {
    * neat way using recursion. 
    */
   def tails[T](l: List[T]): List[List[T]] = {
-    error("fix me")
+    if (l.length == 0) List(l)
+    else List(l) ++ tails[T](l.tail)
```

Comment:
I think you can write `tails[T](l.tail)` as just `tails(l.tail)`, because Scala can infer the type of the argument.